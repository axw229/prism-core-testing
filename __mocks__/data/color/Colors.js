// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
