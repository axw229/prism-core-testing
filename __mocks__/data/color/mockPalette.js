const mockPalette = {
  'colors': [
    {
      'colorNumber': '6765',
      'coordinatingColors': {
        'coord2ColorId': '11194',
        'coord1ColorId': '2870',
        'whiteColorId': '2162'
      },
      'description': [
        'Fairly Colorful',
        'Moderately Colorful',
        'Pale'
      ],
      'id': '2452',
      'isExterior': true,
      'isInterior': true,
      'name': 'Spa',
      'lrv': 64.476,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Blue'
      ],
      'brandKey': 'SW',
      'red': 167,
      'green': 220,
      'blue': 220,
      'hue': 0.5,
      'saturation': 0.43089430894308944,
      'lightness': 0.7588235294117647,
      'hex': '#a7dcdc',
      'isDark': false,
      'storeStripLocator': '164-C2',
      'similarColors': [
        '2634',
        '2630',
        '11225',
        '2472',
        '2445',
        '2622',
        '2459',
        '1606',
        '11226',
        '2451'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '6860',
      'coordinatingColors': {
        'coord2ColorId': '11339',
        'coord1ColorId': '11364',
        'whiteColorId': '2681'
      },
      'description': [
        'Soft',
        'Fairly Bright'
      ],
      'id': '2547',
      'isExterior': true,
      'isInterior': true,
      'name': 'Eros Pink',
      'lrv': 18.874,
      'brandedCollectionNames': [
        '2020 Play',
        '2019 Naturalist'
      ],
      'colorFamilyNames': [
        'Red'
      ],
      'brandKey': 'SW',
      'red': 200,
      'green': 79,
      'blue': 104,
      'hue': 0.9655647382920111,
      'saturation': 0.5238095238095238,
      'lightness': 0.5470588235294118,
      'hex': '#c84f68',
      'isDark': false,
      'storeStripLocator': '105-C5',
      'similarColors': [
        '2273',
        '2266',
        '2542',
        '2546',
        '1998',
        '2280',
        '2530',
        '2287',
        '2279',
        '30193'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9018',
      'coordinatingColors': {
        'coord2ColorId': '11218',
        'coord1ColorId': '11200',
        'whiteColorId': '2366'
      },
      'description': [
        'Intense',
        'Pale'
      ],
      'id': '11195',
      'isExterior': false,
      'isInterior': true,
      'name': 'Honey Bees',
      'lrv': 70.14,
      'brandedCollectionNames': [
        '2018 Unity'
      ],
      'colorFamilyNames': [
        'Yellow'
      ],
      'brandKey': 'SW',
      'red': 251,
      'green': 214,
      'blue': 130,
      'hue': 0.11570247933884299,
      'saturation': 0.937984496124031,
      'lightness': 0.7470588235294118,
      'hex': '#fbd682',
      'isDark': false,
      'storeStripLocator': '134-C4',
      'similarColors': [
        '2382',
        '11199',
        '11196',
        '2588',
        '2375',
        '11198',
        '11194',
        '11191',
        '3146',
        '2361'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9007',
      'coordinatingColors': {
        'coord2ColorId': '1666',
        'coord1ColorId': '2686',
        'whiteColorId': '2331'
      },
      'description': [
        'Rich',
        'Vibrant',
        'Pale'
      ],
      'id': '11184',
      'isExterior': false,
      'isInterior': true,
      'name': 'Melón Meloso',
      'lrv': 55.314,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Orange'
      ],
      'brandKey': 'SW',
      'red': 242,
      'green': 184,
      'blue': 140,
      'hue': 0.0718954248366013,
      'saturation': 0.7968749999999999,
      'lightness': 0.7490196078431373,
      'hex': '#f2b88c',
      'isDark': false,
      'storeStripLocator': '121-C3',
      'similarColors': [
        '2334',
        '2983',
        '2327',
        '11188',
        '2326',
        '2045',
        '2038',
        '3012',
        '2341',
        '2031'
      ],
      'ignore': false,
      'archived': false
    }
  ],
  'activeColor': {
    'colorNumber': '9007',
    'coordinatingColors': {
      'coord2ColorId': '1666',
      'coord1ColorId': '2686',
      'whiteColorId': '2331'
    },
    'description': [
      'Rich',
      'Vibrant',
      'Pale'
    ],
    'id': '11184',
    'isExterior': false,
    'isInterior': true,
    'name': 'Melón Meloso',
    'lrv': 55.314,
    'brandedCollectionNames': [],
    'colorFamilyNames': [
      'Orange'
    ],
    'brandKey': 'SW',
    'red': 242,
    'green': 184,
    'blue': 140,
    'hue': 0.0718954248366013,
    'saturation': 0.7968749999999999,
    'lightness': 0.7490196078431373,
    'hex': '#f2b88c',
    'isDark': false,
    'storeStripLocator': '121-C3',
    'similarColors': [
      '2334',
      '2983',
      '2327',
      '11188',
      '2326',
      '2045',
      '2038',
      '3012',
      '2341',
      '2031'
    ],
    'ignore': false,
    'archived': false
  },
  'toggleCompareColor': false,
  'temporaryActiveColor': null
}

export default mockPalette
