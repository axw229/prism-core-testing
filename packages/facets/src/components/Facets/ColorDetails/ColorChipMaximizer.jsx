// @flow
import type { Node } from 'react'
import React, { type Element, createRef, useContext, useEffect, useState } from 'react'
import { LiveMessage } from 'react-aria-live'
import { FormattedMessage, useIntl } from 'react-intl'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as GA from 'src/analytics/GoogleAnalytics'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'
import ConfigurationContext, {
  type ConfigurationContextType
} from 'src/contexts/ConfigurationContext/ConfigurationContext'
import 'src/providers/fontawesome/fontawesome'
import useChipMaximizer, { BASE_CLASS } from '../../../hooks/useChipMaximizer'
import { type Color } from '../../../shared/types/Colors.js.flow'
import 'src/scss/convenience/visually-hidden.scss'

type Props = {
  addColorBtn: ({ style: $Shape<CSSStyleDeclaration> }) => Element<any>,
  color: Color,
  onToggle?: (boolean) => void,
  isMaximized: boolean,
  setMaximized: (boolean) => void
}

const ColorChipMaximizer = ({ addColorBtn, color, onToggle, isMaximized, setMaximized }: Props): Node => {
  const { brandId }: ConfigurationContextType = useContext(ConfigurationContext)
  const [liveRegionMessage, setLiveRegionMessage] = useState('')
  const { formatMessage, locale } = useIntl()
  const { styles, contrastingTextColor } = useChipMaximizer({ isDark: color.isDark, isMaximized })

  const toggleMaximizedBtn = createRef()

  const toggleChipMaximized = (e) => {
    setLiveRegionMessage('')
    setMaximized(!isMaximized)
  }

  useEffect(() => {
    onToggle && onToggle(isMaximized)
    if (isMaximized) {
      toggleMaximizedBtn.current && toggleMaximizedBtn.current.focus()
      setTimeout(() => {
        setLiveRegionMessage(formatMessage({ id: 'CHIP_MAXIMIZED' }))
      }, 500)
      GA.event(
        {
          category: 'Color Detail',
          action: 'Maximize Swatch',
          label: 'Maximize Swatch'
        },
        GA_TRACKER_NAME_BRAND[brandId]
      )
    }
    if (!isMaximized) {
      toggleMaximizedBtn.current && toggleMaximizedBtn.current.focus()
      setTimeout(() => {
        setLiveRegionMessage(formatMessage({ id: 'CHIP_MINIMIZED' }))
      }, 500)
    }
  }, [isMaximized])

  return (
    <>
      <div className={styles.chip} style={{ backgroundColor: color.hex, zIndex: isMaximized ? 3 : 0 }}>
        {addColorBtn(isMaximized ? {} : { style: { display: 'none' } })}
      </div>
      <div className={styles.wrapper}>
        <button
          className={isMaximized ? styles.alt : styles.swatch}
          onClick={toggleChipMaximized}
          ref={toggleMaximizedBtn}
        >
          <FontAwesomeIcon
            className={`${BASE_CLASS}__display-toggles-icon`}
            icon={isMaximized ? ['fal', 'compress-alt'] : ['fal', 'expand-alt']}
            color={contrastingTextColor}
            size={'2x'}
          />
          <div className={`${BASE_CLASS}__scene-toggle-copy visually-hidden`}>
            {isMaximized ? (
              <FormattedMessage id='RESTORE_COLOR_SWATCH_TO_DEFAULT_SIZE' />
            ) : (
              <FormattedMessage id='MAXIMIZE_COLOR_SWATCH' />
            )}
          </div>
        </button>
      </div>

      <LiveMessage message={liveRegionMessage} aria-live='polite' clearOnUnmount='true' />
    </>
  )
}

export default ColorChipMaximizer
