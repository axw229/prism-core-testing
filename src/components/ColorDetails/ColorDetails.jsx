// @flow
import React, { PureComponent, Fragment } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { flatten, filter } from 'lodash'

import ColorViewer from './ColorViewer/ColorViewer'
import ColorStrip from './ColorStrip/ColorStrip'
import CoordinatingColors from './CoordinatingColors/CoordinatingColors'
import SimilarColors from './SimilarColors/SimilarColors'
import SceneManager from '../SceneManager/SceneManager'

class ColorDetails extends PureComponent<Props> {
  render () {
    const { match: { params }, colors } = this.props

    // grab the color by color number from the URL
    const activeColor = this.getColorByColorNumber(params.colorNumber)

    return (
      <Fragment>
        <h3>Color Details Pane</h3>
        <ColorViewer color={activeColor} />
        <ColorStrip key={activeColor.id} colors={colors} color={activeColor} />
        <CoordinatingColors colors={colors} color={activeColor} />
        <SimilarColors colors={colors} color={activeColor} />
        <SceneManager />
      </Fragment>
    )
  }

  getColorByColorNumber (colorNumber) {
    const { colors } = this.props
    const color = filter(colors, color => color.colorNumber === colorNumber)[0]

    if (!color) {
      return null
    }

    return color
  }
}

const mapStateToProps = (state, props) => {
  const { colors } = state

  // TODO: WhooWhee, don't keep this. Need this here to be able to easily grab a color byId.
  const colours = flatten(Object.keys(colors.items).map(fam => colors.items[fam]))

  return {
    colors: colours
  }
}

export default withRouter(connect(mapStateToProps, null)(ColorDetails))
