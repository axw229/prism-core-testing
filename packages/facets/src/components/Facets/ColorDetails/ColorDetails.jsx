// @flow
import React, { useContext, useEffect, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import { useDispatch } from 'react-redux'
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs'
import { cloneDeep } from 'lodash/lang'
import * as GA from 'src/analytics/GoogleAnalytics'
import ColorWallContext from 'src/components/Facets/ColorWall/ColorWallContext'
import HeroLoader from 'src/components/Loaders/HeroLoader/HeroLoader'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'
import ConfigurationContext, {
  type ConfigurationContextType
} from 'src/contexts/ConfigurationContext/ConfigurationContext'
import ColorDataWrapper from 'src/helpers/ColorDataWrapper/ColorDataWrapper'
import type { SceneStatus } from 'src/shared/types/Scene'
import 'src/providers/fontawesome/fontawesome'
import type { Color } from '../../../shared/types/Colors.js.flow'
import { toggleColorDetailsPage } from '../../../store/actions/scenes'
import { Content } from '../ColorWall/ColorSwatch/ColorSwatch'
import ColorChipMaximizer from './ColorChipMaximizer'
import { type ColorDetailsCTAData, ColorDetailsCTAs } from './ColorDetailsCTAs'
import ColorDetailsScenes from './ColorDetailsScenes'
import ColorInfo from './ColorInfo'
import ColorStrip from './ColorStrip'
import ColorViewer from './ColorViewer'
import CoordinatingColors from './CoordinatingColors'
import SimilarColors from './SimilarColors'
import 'src/scss/convenience/visually-hidden.scss'
import './ColorDetails.scss'

const baseClass = 'color-info'
const mainInfoClass = `${baseClass}__main-info`

type Props = {
  onColorChanged?: (Color) => void,
  onSceneChanged?: (SceneStatus) => void,
  onVariantChanged?: (string) => void,
  onColorChipToggled?: (boolean) => void,
  familyLink?: string,
  loading?: boolean,
  colorFromParent?: Color,
  initialVariantName?: string,
  callsToAction?: ColorDetailsCTAData[]
}

type ColorMessage = {
  ...Color,
  doNotBroadcast?: boolean
}

export const ColorDetails = ({
  onColorChanged,
  onSceneChanged,
  onVariantChanged,
  onColorChipToggled,
  familyLink,
  loading,
  colorFromParent = {},
  initialVariantName,
  callsToAction = []
}: Props) => {
  const dispatch = useDispatch()
  const { brandId }: ConfigurationContextType = useContext(ConfigurationContext)
  const [color, setColor] = useState<ColorMessage>(colorFromParent) // in some case we add an extra prop to a color obj that tells the pub method to not fire
  const [tabIndex, setTabIndex] = useState<number>(0)
  const [isMaximized, setMaximized] = useState(false)

  useEffect(() => {
    if (colorFromParent && colorFromParent.id !== color.id) {
      const _color = cloneDeep(colorFromParent)
      _color.doNotBroadcast = true
      setColor(_color)
    }
  }, [colorFromParent])

  useEffect(() => {
    dispatch(toggleColorDetailsPage())
    return () => {
      dispatch(toggleColorDetailsPage())
    }
  }, [])

  useEffect(() => {
    if (color) {
      GA.pageView(`color-detail/${color.brandKey} ${color.colorNumber} - ${color.name}`, GA_TRACKER_NAME_BRAND[brandId])
    }
    if (onColorChanged && !color?.doNotBroadcast) {
      onColorChanged(color)
    }
    // force tab change to first tab if there is no coordinating colors tab
    color.coordinatingColors || setTabIndex(0)
  }, [color])

  const AddColorBtn = ({ style }: { style?: $Shape<CSSStyleDeclaration> }) => {
    const { colorDetailsAddColor } = useContext<ConfigurationContextType>(ConfigurationContext)

    if (!colorDetailsAddColor) return null

    return (
      <div className={`${mainInfoClass}--add`} style={style}>
        <ColorWallContext.Provider value={{ displayAddButton: true }}>
          <Content msg='' color={color} isMaximized={isMaximized} style={{ position: 'relative' }} />
        </ColorWallContext.Provider>
      </div>
    )
  }

  if (loading) {
    return <HeroLoader />
  }

  return (
    <>
      <div className='color-detail-view'>
        <ColorChipMaximizer
          addColorBtn={(style) => <AddColorBtn style={style} isMaximized={isMaximized} />}
          color={color}
          isMaximized={isMaximized}
          setMaximized={setMaximized}
          onToggle={onColorChipToggled}
        />
        <div className={'color-detail__scene-wrapper color-detail__scene-wrapper--displayed'}>
          <ColorDetailsScenes color={color} isMaximized={isMaximized} intitialVariantName={initialVariantName} />
        </div>
        <div className='color-detail__info-wrapper'>
          <div
            className={`${mainInfoClass}${color.isDark ? ` ${mainInfoClass}--dark-color` : ''}`}
            style={{ backgroundColor: color.hex }}
          >
            <ColorViewer color={color} />
            <ColorStrip key={color.id} color={color} onColorChanged={setColor} />
            {!isMaximized ? <AddColorBtn /> : null}
          </div>

          {callsToAction?.length ? (
            <ColorDetailsCTAs className='color-detail__ctas color-detail__ctas--mobile' data={callsToAction} />
          ) : null}

          <div className={`${baseClass}__additional-info`}>
            <Tabs
              selectedIndex={tabIndex}
              onSelect={(index) => {
                const tabNames = ['View Coord Color Section', 'View Similar Color Section', 'View Color Info Section']
                GA.event(
                  { category: 'Color Detail', action: tabNames[index], label: tabNames[index] },
                  GA_TRACKER_NAME_BRAND[brandId]
                )
                setTabIndex(index)
              }}
            >
              <TabList className={`${baseClass}__tab-list`} style={{ backgroundColor: color.hex }}>
                {color?.coordinatingColors?.coord1ColorId && (
                  <Tab
                    className={`coordinating-colors-tab ${baseClass}__tab ${
                      color.isDark ? `${baseClass}__tab--dark-color` : ''
                    }`}
                  >
                    <div className={`${baseClass}__tab-copy`}>
                      <FormattedMessage id='COORDINATING_COLORS' />
                    </div>
                  </Tab>
                )}
                <Tab
                  className={`similar-colors-tab ${baseClass}__tab ${
                    color.isDark ? `${baseClass}__tab--dark-color` : ''
                  }`}
                >
                  <div className={`${baseClass}__tab-copy`}>
                    <FormattedMessage id='SIMILAR_COLORS' />
                  </div>
                </Tab>
                <Tab
                  className={`color-info-tab ${baseClass}__tab ${color.isDark ? `${baseClass}__tab--dark-color` : ''}`}
                >
                  <div className={`${baseClass}__tab-copy`}>
                    <FormattedMessage id='DETAILS' />
                  </div>
                </Tab>
              </TabList>
              {color?.coordinatingColors?.coord1ColorId && (
                <TabPanel className={`${baseClass}__tab-panel`}>
                  <CoordinatingColors color={color} onColorChanged={setColor} />
                </TabPanel>
              )}
              <TabPanel className={`${baseClass}__tab-panel`}>
                <SimilarColors color={color} onColorChanged={setColor} />
              </TabPanel>
              <TabPanel className={`${baseClass}__tab-panel color-info__tab-panel-details`}>
                <ColorInfo color={color} familyLink={familyLink} />
              </TabPanel>
            </Tabs>
          </div>
        </div>
      </div>

      {callsToAction?.length ? (
        <ColorDetailsCTAs className='color-detail__ctas color-detail__ctas--desktop' data={callsToAction} />
      ) : null}
    </>
  )
}

export default ColorDataWrapper(ColorDetails)
