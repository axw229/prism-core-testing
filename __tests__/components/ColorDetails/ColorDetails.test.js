import React from 'react'
import { shallow } from 'enzyme'
import { ColorDetails } from 'src/components/Facets/ColorDetails/ColorDetails'
import * as Colors from '__mocks__/data/color/Colors'

// init props of Compoennt
const color = Colors.getColor()
const match = { params: { colorId: 'id', router: jest.fn() } }
const config = { ga_domain_id: 'sherwinWilliamsDefault' }

const wrapper = shallow(<ColorDetails match={match} colors={color} config={config} />)

// Snapshot testing
describe('Color Details Container Component should rendering correctly', () => {
  it('it should match snapshot and rendering corretly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
// Rendering testing
describe('Testing props pass ccorrectly for color details component', () => {
  it('ColorChipMaximizer component should rendering when get correct props', () => {
    expect(wrapper.find('div.color-detail-view:first-child').exists()).toBe(true)
  })

  it('ColorViewer component should rendering when get correct props', () => {
    expect(wrapper.find('ColorViewer').exists()).toBe(true)
  })

  it('SceneManager component should rendering when get correct props', () => {
    expect(wrapper.find('div.color-detail__scene-wrapper').exists()).toBe(true)
  })

  it('ColorStrip component should rendering when get correct props', () => {
    expect(wrapper.find('div.color-info__main-info').exists()).toBe(true)
  })

  it('Coordinating Colors component should rendering when get correct props', () => {
    expect(wrapper.find('CoordinatingColors').exists()).toBe(true)
  })

  it('SimilarColors component should rendering when get correct props', () => {
    expect(wrapper.find('SimilarColors').exists()).toBe(true)
  })

  it('Color Info component should rendering when get correct props', () => {
    expect(wrapper.find('ColorInfo').exists()).toBe(true)
  })
})

describe('Testing state for color details component', () => {
  it('State should initialize correctly', () => {
    const detailsState = wrapper.instance().state
    expect(detailsState).toEqual({
      sceneIsDisplayed: true,
      chipIsMaximized: false,
      a11yMessage: '',
      a11yAssertMessage: ''
    })
  })
})

// Behavior testing
describe('Event tesing for behavoir of color details component', () => {
  it('Testing for toggle Scene Display ', () => {
    const toggleSceneDisplayMock = jest.fn()
    wrapper.instance().toggleSceneDisplay = toggleSceneDisplayMock
    wrapper.instance().forceUpdate()
    wrapper.find('div.color-detail__info-wrapper > button:first-child').simulate('click')
    expect(toggleSceneDisplayMock).toHaveBeenCalled()
  })

  it('Testing for report tabSwitch to GA', () => {
    const reportTabSwitchToGAMock = jest.fn()
    wrapper.instance().reportTabSwitchToGA = reportTabSwitchToGAMock
    wrapper.instance().forceUpdate()
    wrapper.find('Tabs').simulate('select')
    expect(reportTabSwitchToGAMock).toHaveBeenCalled()
  })
})
