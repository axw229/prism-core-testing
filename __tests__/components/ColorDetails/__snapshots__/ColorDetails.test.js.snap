// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Details Container Component should rendering correctly it should match snapshot and rendering corretly 1`] = `
<Fragment>
  <div
    className="color-detail-view"
  >
    <InjectIntl(ColorChipMaximizer)
      color="2248"
    />
    <div
      className="color-detail__scene-wrapper  color-detail__scene-wrapper--displayed"
    >
      <Connect(SceneManager)
        interactive={false}
        mainColor="2248"
        maxActiveScenes={1}
      />
    </div>
    <div
      className="color-detail__info-wrapper"
    >
      <button
        className="color-info__display-toggle-button color-info__scene-display-toggle-button color-info__display-toggle-button--active"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="color-info__display-toggles-icon color-info__display-toggles-icon--scene"
          color="#000000"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "home",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <div
          className="color-info__scene-toggle-copy"
        >
          <FormattedMessage
            id="DISPLAY_SCENE_PAINTER"
            values={Object {}}
          />
        </div>
      </button>
      <button
        className="color-info__display-toggle-button color-info__scene-display-toggle-button color-info__scene-display-toggle-button--alt color-info__display-toggle-button--active"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="color-info__display-toggles-icon color-info__display-toggles-icon--scene"
          color="#000000"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fas",
              "home",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <div
          className="color-info__scene-toggle-copy"
        >
          <FormattedMessage
            id="HIDE_SCENE_PAINTER"
            values={Object {}}
          />
        </div>
      </button>
      <div
        className="color-info__main-info"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      >
        <ColorViewer
          color="2248"
        />
        <withRouter(ColorStrip)
          color="2248"
          colors={
            Object {
              "blue": 219,
              "brandKey": "SW",
              "colorFamilyNames": Array [
                "Red",
              ],
              "colorNumber": "6561",
              "coordinatingColors": Object {
                "coord1ColorId": 2250,
                "coord2ColorId": 1926,
                "whiteColorId": 2681,
              },
              "cssrgb": "rgb(235,209,219)",
              "description": Array [
                "Subdued",
                "Muted",
                "Brilliant",
                "Radiant",
              ],
              "green": 209,
              "hex": "#ebd1db",
              "hue": 0.935897435897436,
              "id": "2248",
              "isDark": false,
              "isExterior": true,
              "isInterior": true,
              "lightness": 0.8705882352941177,
              "lrv": 68.551,
              "name": "Teaberry",
              "red": 235,
              "rgb": 15454683,
              "saturation": 0.39393939393939387,
              "similarColors": Array [
                "2255",
                "2262",
                "2269",
                "2662",
                "2535",
                "2276",
                "2658",
                "1974",
                "2543",
                "2531",
              ],
              "storeStripLocator": "102-C1",
            }
          }
        />
      </div>
      <div
        className="color-info__additional-info"
      >
        <Tabs
          defaultFocus={false}
          defaultIndex={null}
          forceRenderTabPanel={false}
          onSelect={[Function]}
          selectedIndex={null}
        >
          <TabList
            className="color-info__tab-list"
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
          >
            <Tab
              className="color-info__tab "
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id={null}
              panelId={null}
              selected={false}
              selectedClassName="react-tabs__tab--selected"
            >
              <div
                className="color-info__tab-copy"
              >
                <FormattedMessage
                  id="COORDINATING_COLORS"
                  values={Object {}}
                />
              </div>
            </Tab>
            <Tab
              className="color-info__tab "
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id={null}
              panelId={null}
              selected={false}
              selectedClassName="react-tabs__tab--selected"
            >
              <div
                className="color-info__tab-copy"
              >
                <FormattedMessage
                  id="SIMILAR_COLORS"
                  values={Object {}}
                />
              </div>
            </Tab>
            <Tab
              className="color-info__tab "
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id={null}
              panelId={null}
              selected={false}
              selectedClassName="react-tabs__tab--selected"
            >
              <div
                className="color-info__tab-copy"
              >
                <FormattedMessage
                  id="DETAILS"
                  values={Object {}}
                />
              </div>
            </Tab>
          </TabList>
          <TabPanel
            className="color-info__tab-panel"
            forceRender={false}
            selectedClassName="react-tabs__tab-panel--selected"
          >
            <CoordinatingColors
              color="2248"
              colors={
                Object {
                  "blue": 219,
                  "brandKey": "SW",
                  "colorFamilyNames": Array [
                    "Red",
                  ],
                  "colorNumber": "6561",
                  "coordinatingColors": Object {
                    "coord1ColorId": 2250,
                    "coord2ColorId": 1926,
                    "whiteColorId": 2681,
                  },
                  "cssrgb": "rgb(235,209,219)",
                  "description": Array [
                    "Subdued",
                    "Muted",
                    "Brilliant",
                    "Radiant",
                  ],
                  "green": 209,
                  "hex": "#ebd1db",
                  "hue": 0.935897435897436,
                  "id": "2248",
                  "isDark": false,
                  "isExterior": true,
                  "isInterior": true,
                  "lightness": 0.8705882352941177,
                  "lrv": 68.551,
                  "name": "Teaberry",
                  "red": 235,
                  "rgb": 15454683,
                  "saturation": 0.39393939393939387,
                  "similarColors": Array [
                    "2255",
                    "2262",
                    "2269",
                    "2662",
                    "2535",
                    "2276",
                    "2658",
                    "1974",
                    "2543",
                    "2531",
                  ],
                  "storeStripLocator": "102-C1",
                }
              }
            />
          </TabPanel>
          <TabPanel
            className="color-info__tab-panel"
            forceRender={false}
            selectedClassName="react-tabs__tab-panel--selected"
          >
            <SimilarColors
              color="2248"
              colors={
                Object {
                  "blue": 219,
                  "brandKey": "SW",
                  "colorFamilyNames": Array [
                    "Red",
                  ],
                  "colorNumber": "6561",
                  "coordinatingColors": Object {
                    "coord1ColorId": 2250,
                    "coord2ColorId": 1926,
                    "whiteColorId": 2681,
                  },
                  "cssrgb": "rgb(235,209,219)",
                  "description": Array [
                    "Subdued",
                    "Muted",
                    "Brilliant",
                    "Radiant",
                  ],
                  "green": 209,
                  "hex": "#ebd1db",
                  "hue": 0.935897435897436,
                  "id": "2248",
                  "isDark": false,
                  "isExterior": true,
                  "isInterior": true,
                  "lightness": 0.8705882352941177,
                  "lrv": 68.551,
                  "name": "Teaberry",
                  "red": 235,
                  "rgb": 15454683,
                  "saturation": 0.39393939393939387,
                  "similarColors": Array [
                    "2255",
                    "2262",
                    "2269",
                    "2662",
                    "2535",
                    "2276",
                    "2658",
                    "1974",
                    "2543",
                    "2531",
                  ],
                  "storeStripLocator": "102-C1",
                }
              }
            />
          </TabPanel>
          <TabPanel
            className="color-info__tab-panel color-info__tab-panel-details"
            forceRender={false}
            selectedClassName="react-tabs__tab-panel--selected"
          >
            <ColorInfo
              color="2248"
            />
          </TabPanel>
        </Tabs>
      </div>
    </div>
  </div>
  <LiveMessage
    aria-live="polite"
    message="You are now on the undefined color detail page."
  />
  <LiveMessage
    aria-live="polite"
    clearOnUnmount="true"
    message=""
  />
  <LiveMessage
    aria-live="assertive"
    clearOnUnmount="true"
    message=""
  />
</Fragment>
`;
